partner_ref_unique/tests/test_res_partner_ref.py

```diff
@@ -9,49 +10,83 @@ class TestResPartnerRefUnique(common.SavepointCase):
     @classmethod
     def setUpClass(cls):
         super(TestResPartnerRefUnique, cls).setUpClass()
-        cls.company = cls.env.ref("base.main_company")
         cls.partner_obj = cls.env["res.partner"]
-        cls.partner1 = cls.partner_obj.create({"name": "Partner1",})
-        cls.partner2 = cls.partner_obj.create({"name": "Partner2",})
+        cls.company_obj = cls.env["res.company"]
+        # look for possible already duplicated refs for being resilient
+        groups = cls.partner_obj.read_group([], ["ref"], ["ref"],)
+        for group in groups:
+            if group["ref_count"] > 1 and not group["ref"]:
+                cls.partner_obj.search(group["__domain"]).write({"ref": False})
```

Comment:
We talked about resetting all refs:
```suggestion
        groups = cls.partner_obj.search([("ref", "!=", False)]).write({"ref": False})
```